util/image_viewer/spotfinder_frame.py

```diff
@@ -1014,6 +1015,15 @@ def update_settings(self, layout=True):
         ):
             self.pyslip.tiles.update_brightness(new_brightness, new_color_scheme)
 
+        detector = self.pyslip.tiles.raw_image.get_detector()
+        if self.params.project_onto == "image":
+            if not hasattr(detector, "projected_2d"):
+                detector.projected_2d = project_2d(detector)
+            elif detector.projected_2d is None:
+                detector.projected_2d = project_2d(detector)
```

Comment:
```suggestion
            if not getattr(detector, "projected_2d", None):
                detector.projected_2d = project_2d(detector)
```

Reply:
☝️  this is kinda what I was getting at 🙂

I appreciate I am not always clear though